### PR TITLE
Add Qwen3 device top-k candidate selector

### PR DESCRIPTION
**Pull request by zmnobug**
## Summary

Add a standalone Qwen3-14B NPU operator that selects the top 32 sampling candidates from padded full-vocabulary logits.

This is the pypto-lib side of extending the device sampling path introduced by hw-native-sys/pypto-serving#47 from greedy sampling to top-k sampling.

## What Changed

- Add `models/qwen3/14b/topk_select.py`.
- Select four local candidates from each 64-token vocabulary chunk.
- Merge the chunk candidates and return the global top 32 values and token IDs.
- Exclude padded vocabulary entries using `REAL_VOCAB` and tail-chunk masking.
- Support the runtime active batch count while preserving fixed compile-time tensor shapes.
- Add a PyTorch golden implementation and pair-aware output comparison.

## Integration

The paired pypto-serving change compiles this operator as a separate L3 callable and uses its small candidate output for host-side temperature, top-p, and multinomial sampling.

Tracking issue: https://github.com/hw-native-sys/pypto-serving/issues/74

## Validation

- Integrated Qwen3-14B top-k generation completed on `a2a3` with `temperature=0.8` and `top_k=32`.
- The paired serving source and unit tests for device candidate routing pass locally.

### CODE REVIEW

File: models/qwen3/14b/topk_select.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+@pl.jit
+def topk_select_fwd(
+    logits: pl.Tensor[[BATCH, VOCAB], pl.FP32],
+    active_batch: pl.Tensor[[1], pl.INT32],
+    topk_values: pl.Out[pl.Tensor[[BATCH, TOPK], pl.FP32]],
+    topk_indices: pl.Out[pl.Tensor[[BATCH, TOPK], pl.INT32]],
+):
+    """Return the largest TOPK real-vocab logits and token ids per batch row."""
+    for b in pl.parallel(BATCH):
+        if b < pl.read(active_batch, [0]):
```
[the remark this answers is not in the document]
Comment (zmnobug) [reply models/qwen3/14b/topk_select.py]: Not applied. sampling_control is a device tensor consumed under pl.parallel; hoisting pl.read outside that construct does not provide a verified cross-core shared load in the PyPTO execution model and may change lowering/scope semantics. The current path performs one 32-bit control read per active parallel row/core, which is negligible beside the vocabulary sorting work, so we keep the established in-loop pattern.

File: models/qwen3/14b/topk_select.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                for k in pl.range(TOPK):
+                    pl.write(topk_values, [b, k], pl.read(acc_vals, [0, k]))
+                    candidate_pos = pl.read(selected_positions, [0, k])
+                    token_id = pl.read(candidate_ids, [0, pl.cast(candidate_pos, target_type=pl.INDEX)])
+                    pl.write(topk_indices, [b, k], token_id)
+        else:
+            with pl.at(level=pl.Level.CORE_GROUP, name_hint="topk_select_inactive"):
+                for k in pl.range(TOPK):
+                    pl.write(topk_values, [b, k], pl.cast(FP32_NEG_INF, target_type=pl.FP32))
+                    pl.write(topk_indices, [b, k], pl.cast(0, target_type=pl.INT32))
```
[the remark this answers is not in the document]
Comment (zmnobug) [reply models/qwen3/14b/topk_select.py]: Addressed in 0027d67. Inactive value and index rows, as well as greedy output initialization, now use pl.full with vector slice assignment. Greedy and top-k device tests both pass on a2a3.

File: models/qwen3/14b/topk_select.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                    candidate_pairs,
+                    mask_pattern=pl.tile.MaskPattern.P1010,
+                    output_dtype=pl.INT32,
+                )
+
+                for k in pl.range(TOPK):
+                    pl.write(topk_values, [b, k], pl.read(acc_vals, [0, k]))
+                    candidate_pos = pl.read(selected_positions, [0, k])
+                    token_id = pl.read(candidate_ids, [0, pl.cast(candidate_pos, target_type=pl.INDEX)])
+                    pl.write(topk_indices, [b, k], token_id)
```
[the remark this answers is not in the document]
Comment (zmnobug) [reply models/qwen3/14b/topk_select.py]: Addressed in 0027d67. topk_values is now written with one vector slice assignment instead of 32 scalar reads/writes. Both selection-k=1 and selection-k=32 passed golden comparison on a2a3 NPU.

**Comment by zmnobug** [comment]
Addressed the automated review feedback in `0027d67`:\n\n- Replaced scalar `topk_values` writes with a vector slice assignment.\n- Replaced inactive-row and greedy initialization loops with `pl.full` vector writes.\n- Kept `sampling_control` reads inside `pl.parallel`: hoisting a device-tensor read does not provide verified cross-core sharing in the PyPTO execution model and may change lowering scope.\n\nBoth `selection-k=1` and `selection-k=32` passed golden comparison on a2a3 NPU; Ruff checks also passed.
